neqo-transport/src/connection.rs

```diff
@@ -5173,4 +5187,131 @@ mod tests {
         client.process_input(Datagram::new(loopback(), loopback(), vec![77; 21]), now());
         assert!(matches!(client.state(), State::Draining { .. }));
     }
+
+    #[test]
+    fn server_receive_unknown_first_packet() {
+        let mut server = default_server();
+
+        let mut unknown_version_packet = vec![0x80, 0x1a, 0x1a, 0x1a, 0x1a];
+        unknown_version_packet.resize(1200, 0x0);
+
+        assert_eq!(
+            server.process(
+                Some(Datagram::new(
+                    loopback(),
+                    loopback(),
+                    unknown_version_packet,
+                )),
+                now(),
+            ),
+            Output::None
```

Comment:
Ahh, the reason that this is not generating a VN response is that we generate that in Server not Connection.  That took me longer to realize than I should be happy to admit.  Could you add a comment explaining that?